src/main/java/br/unb/cic/analysis/SootWrapper.java

```diff
@@ -46,20 +45,32 @@ private static List<String> getIncludeList() {
         return stringList;
     }
 
-    public static void configureSootOptionsToRunInterproceduralOverrideAssignmentAnalysis(List<String> classpath) {
-        soot.options.Options.v().set_no_bodies_for_excluded(true);
-        soot.options.Options.v().set_allow_phantom_refs(true);
-        soot.options.Options.v().set_output_format(soot.options.Options.output_format_jimple);
-        soot.options.Options.v().set_whole_program(true);
-        soot.options.Options.v().set_process_dir(classpath);
-        soot.options.Options.v().set_full_resolver(true);
-        soot.options.Options.v().set_keep_line_number(true);
-        soot.options.Options.v().set_prepend_classpath(false);
-        soot.options.Options.v().set_include(getIncludeList());
+    public static void configureSootOptionsToRunInterproceduralOverrideAssignmentAnalysis(String classpath) {
+        List<String> classes = Collections.singletonList(classpath);
+
+        Options.v().set_no_bodies_for_excluded(true);
+        Options.v().set_allow_phantom_refs(true);
+        Options.v().set_output_format(soot.options.Options.output_format_jimple);
+        Options.v().set_whole_program(true);
+        Options.v().set_process_dir(classes);
+        Options.v().set_full_resolver(true);
+        Options.v().set_keep_line_number(true);
+        Options.v().set_include(getIncludeList());
+
+        // JAVA 8
+        if (getJavaVersion() < 9) {
+            Options.v().set_prepend_classpath(true);
+            Options.v().set_soot_classpath(classpath + File.pathSeparator + pathToJCE() + File.pathSeparator + pathToRT());
+        }
+        // JAVA VERSION 9 && IS A CLASSPATH PROJECT
+        else if (getJavaVersion() >= 9) {
+            Options.v().set_soot_classpath(classpath);
```

Comment:
Aqui a string `VIRTUAL_FS_FOR_JDK` precisa ser passada para o classpath ou o soot não inicia corretamente as classes. ex.:
```
Options.v().set_soot_classpath("VIRTUAL_FS_FOR_JDK" + File.pathSeparator + classpath);
```

Reply:
Esse erro deixa de acontecer ao atualizar a versão da dependência do soot para 4.3.0. A versão utilizada atualmente não suporta esse fix.

Reply:
Entao vamos deixar para fazer esse fix com a atualização do soot em outro PR, ok?

Reply:
@barbosamaatheus você já faz isso então? pelo que entendi essa versão deste PR aqui ficaria bugada mesmo?

Reply:
@pauloborba se for considerar bugado é um bug que sempre existiu. Acho melhor fazer em outro PR porque vamos precisar atualizar o soot e aí faremos melhores testes além de conseguir revisar melhor as mudanças.